### PR TITLE
proto: Not use bazel to generate bad descriptor set

### PR DESCRIPTION
**Pull request by lizan**
The PR is to simplify Google import process during translating bazel genrules for proto descriptors. Also make it consistent to other bad descriptors which are generated within the test.

### CODE REVIEW

File: test/common/grpc/json_transcoder_filter_test.cc [hunk 1]
```diff
@@ -75,6 +76,17 @@ class GrpcJsonTranscoderConfigTest : public testing::Test {
       }
     }
   }
+
+  void stripImports(FileDescriptorSet& descriptor_set, std::string file_name) {
+    FileDescriptorProto file_descriptor;
+    for (auto& file : *descriptor_set.mutable_file()) {
+      if (file.name() == file_name) {
```
Comment (htuch) [inline test/common/grpc/json_transcoder_filter_test.cc]: Nit: can you switch to `std::find_if`?

File: test/common/grpc/json_transcoder_filter_test.cc [hunk 2]
```diff
@@ -75,6 +76,17 @@ class GrpcJsonTranscoderConfigTest : public testing::Test {
       }
     }
   }
+
+  void stripImports(FileDescriptorSet& descriptor_set, std::string file_name) {
+    FileDescriptorProto file_descriptor;
+    for (auto& file : *descriptor_set.mutable_file()) {
```
Comment (htuch) [inline test/common/grpc/json_transcoder_filter_test.cc]: Nit: can you add a comment saying "filter down descriptor_set to only bookstore.proto but none of its dependencies"?

**Comment by dnoe** [comment]
Can you add a little bit more detail about the reason for this change in the PR description? Thanks.

File: test/common/grpc/json_transcoder_filter_test.cc [hunk 3]
```diff
@@ -75,6 +76,20 @@ class GrpcJsonTranscoderConfigTest : public testing::Test {
...
   }
+
+  void stripImports(FileDescriptorSet& descriptor_set, std::string file_name) {
+    FileDescriptorProto file_descriptor;
+    // filter down descriptor_set to only contain one proto specified as file_name but none of its
+    // dependencies
+    auto file_itr = std::find_if(
+        descriptor_set.file().begin(), descriptor_set.file().end(),
+        [&file_name](const FileDescriptorProto& file) { return file.name() == file_name; });
+    if (file_itr != descriptor_set.file().end()) {
```
Comment (htuch) [inline test/common/grpc/json_transcoder_filter_test.cc]: Could just make this a RELEASE_ASSERT, but it's fine in any case.

**Comment by lizan** [comment]
Travis is failing because of "No space left", any idea to solve that?

**Comment by mattklein123** [comment]
@junr03 looks like sometimes "minimal" is not working and it defaults to "ruby" which I think makes us run out of space. Can you ask Travis why this is happening?

**Comment by hennna** [comment]
Is it possible to close/reopen the PR to kickstart Travis?

**Comment by junr03** [comment]
@mattklein123 According to travis engineers "the message is not indicative of how travis-build behaves and interprets the .travis.yml". This was their response when I brought up the message you are seeing, and they opened this PR to fix the messaging https://github.com/travis-ci/travis-yaml/pull/119.

@lizan this means that this PR is ostensibly running in the same environment that all other passing PRs are. Do you have a sense if this PR would cause more disk usage than previous ones?

I have restarted the tsan job to see is the behavior is deterministic.

**Comment by htuch** [comment]
This should use strictly less space than before (although probably by only a few kb).

**Comment by mattklein123** [comment]
This seems to reliably fail. It's been kicked a few times, so something must have changed somehow? Sigh. Do we need to add some debug statements to figure out how much space we are using at the start and after various steps?

**Comment by junr03** [comment]
@mattklein123 yes, I think that at this point that is what we'll have to do.

**Comment by lizan** [comment]
Weird, I can reproduce it on local docker, but not out of docker...

**Comment by lizan** [comment]
Seems it is not related to TSAN or ASAN, just since how bazel was run, under TSAN and ASAN the proto file is built from `external/envoy/test/proto/bookstore.proto` so the path didn't match. The test should be fixed now.

File: ci/ci_steps.sh [hunk 4]
```diff
@@ -6,7 +6,7 @@ set -e
 . ./ci/envoy_build_sha.sh
 
 # Lint travis file.
-travis lint .travis.yml --skip-completion-check
+#travis lint .travis.yml --skip-completion-check
```
Comment (mattklein123) [inline ci/ci_steps.sh]: ?

Comment (lizan) [inline ci/ci_steps.sh]: oops, reverted.

File: test/common/grpc/json_transcoder_filter_test.cc [hunk 5]
```diff
@@ -75,6 +76,25 @@ class GrpcJsonTranscoderConfigTest : public testing::Test {
...
+  void stripImports(FileDescriptorSet& descriptor_set, std::string file_name) {
+    FileDescriptorProto file_descriptor;
+    // filter down descriptor_set to only contain one proto specified as file_name but none of its
+    // dependencies
+    auto file_itr =
+        std::find_if(descriptor_set.file().begin(), descriptor_set.file().end(),
+                     [&file_name](const FileDescriptorProto& file) {
+                       // return whether file.name() ends with file_name
+                       return file.name().length() >= file_name.length() &&
+                              0 == file.name().compare(file.name().length() - file_name.length(),
```
Comment (hennna) [inline test/common/grpc/json_transcoder_filter_test.cc]: ProtobufTypes::FromString(file.name()) here.

Comment (lizan) [inline test/common/grpc/json_transcoder_filter_test.cc]: Ugh, changed `file_name` to `ProtobufTypes::String`

**Comment by hennna** [comment]
LGTM

**Pull request merged**